client/web/src/insights/pages/creation/search-insight/SearchInsightCreationPage.tsx

```diff
@@ -122,27 +127,44 @@ export const SearchInsightCreationPage: React.FunctionComponent<SearchInsightCre
         <Page className={classnames('col-10', styles.creationPage)}>
             <PageTitle title="Create new code insight" />
 
-            <div className="mb-5">
-                <h2>Create new code insight</h2>
-
-                <p className="text-muted">
-                    Search-based code insights analyze your code based on any search query.{' '}
-                    <a href="https://docs.sourcegraph.com/code_insights" target="_blank" rel="noopener">
-                        Learn more.
-                    </a>
-                </p>
-            </div>
-
-            <SearchInsightCreationContent
-                className="pb-5"
-                dataTestId="search-insight-create-page-content"
-                settings={settingsCascade.final}
-                initialValue={initialFormValues}
-                organizations={orgs}
-                onSubmit={handleSubmit}
-                onCancel={handleCancel}
-                onChange={handleChange}
-            />
+            {hasQueryInsight && urlQueryInsightValues === undefined && (
+                // loading state for 1 click creation insight values resolve operation
+                <span>Loading...</span>
+            )}
+
+            {hasQueryInsight && isErrorLike(urlQueryInsightValues) && <ErrorAlert error={urlQueryInsightValues} />}
+
+            {
+                // If we have query in URL we should be sure that we have initial values
+                // from URL query based insight. If we don't have query in URl we can render
+                // page without resolving URL query based insight values.
+                !hasQueryInsight ||
+                    (hasQueryInsight && !isErrorLike(urlQueryInsightValues) && urlQueryInsightValues && (
```

Comment:
We already computed `hasQueryInsight && !isErrorLike(urlQueryInsightValues) && urlQueryInsightValues` value above. Can we use it here too? 

```suggestion
                (!hasQueryInsight || hasUrlQueryInsightValues) && (
```